airbyte-api/src/main/openapi/config.yaml

```diff
@@ -1756,13 +1756,21 @@ components:
           items:
             $ref: "#/components/schemas/SyncMode"
         sourceDefinedCursor:
-          description: If the source defines the cursor field, then it does any other cursor field inputs will be ignored. If it does not either the user_provided one is used or as a backup the default one is used.
+          description: If the source defines the cursor field, then any other cursor field inputs will be ignored. If it does not, either the user_provided one is used, or the default one is used as a backup.
           type: boolean
         defaultCursorField:
           description: Path to the field that will be used to determine if a record is new or modified since the last sync. If not provided by the source, the end user will have to specify the comparable themselves.
           type: array
           items:
             type: string
+        sourceDefinedPrimaryKey:
+          description: If the source defines the primary key, then any other primary key inputs will be ignored. If it does not, either the user_provided one is used, or the default one is used as a backup.
+          type: boolean
```

Comment:
in the `sourceDefinedCursor` case this is a boolean, because we do not need to expose the cursor field outside of the source. All you need to know is it has one. I _think_ it's a bit different in this case though where the primary key has to be exposed outside of the source.

If `sourceDefinedPrimaryKey` is true does that mean `defaultPrimaryKey` will always be set? An alternative is that `sourceDefinedPrimaryKey` instead of being a boolean could actually be a primary key. Up to you on how you want to handle this; I just want to make sure that in the case where the source is defining the primary key that that configuration is handled properly.

Reply:
Ok, thanks for the clarification!